calculator_test.go

```diff
@@ -2,23 +2,212 @@ package calculator_test
+		if testcase.want != got {
+			t.Errorf("Wanted %v, got %v in %v", testcase.want, got, testcase.name)
+		}
+	}
+}
+
+func TestSubtractRandom(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	for i := 0; i < 100; i++ {
+		a, b := rand.Float64(), rand.Float64()
+		want := a - b
+		got := calculator.Subtract(a, b)
+		if want != got {
+			t.Errorf("Want %v, got %v, using %v - %v", want, got, a, b)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	testcases := []struct {
+		a, b, want float64
+		rest       []float64
+		expectsErr bool
+		name       string
+	}{
+		{name: "Multiply two positives", a: 2, b: 2, rest: []float64{}, want: 4},
+		{name: "Multiply positive and unit", a: 2, b: 1, rest: []float64{}, want: 2},
+		{name: "Multiply positive and zero", a: 2, b: 0, rest: []float64{}, want: 0},
+		{name: "Multiply zero and zero", a: 0, b: 0, rest: []float64{}, want: 0},
+		{name: "Multiply zero and negative", a: 0, b: -1, rest: []float64{}, want: 0},
+		{name: "Multiply positive and negative units", a: 1, b: -1, rest: []float64{}, want: -1},
+		{name: "Multiply positive and negative unit", a: 2, b: -1, rest: []float64{}, want: -2},
+		{name: "Multiply two negative units", a: -1, b: -1, rest: []float64{}, want: 1},
+		{name: "Multiply negative and negative unit", a: -2, b: -1, rest: []float64{}, want: 2},
+		{name: "Multiply two negatives", a: -2, b: -2, rest: []float64{}, want: 4},
+		{name: "Multiply two numbers", a: 1, b: 2, rest: []float64{}, want: 2},
+		{name: "Multiply three numbers", a: 1, b: 2, rest: []float64{3}, want: 6},
+		{name: "Multiply four numbers", a: 1, b: 2, rest: []float64{3, 4}, want: 24},
+		{name: "Multiply five numbers", a: -1, b: 2, rest: []float64{-3, 4, -5}, want: -120},
+	}
+	for _, testcase := range testcases {
+		got := calculator.Multiply(testcase.a, testcase.b, testcase.rest...)
+		if testcase.want != got {
+			t.Errorf("Wanted %v, got %v in %v", testcase.want, got, testcase.name)
+		}
+	}
+}
+
+func TestDivide(t *testing.T) {
+	testcases := []struct {
+		a, b       float64
+		rest       []float64
+		want       float64
+		expectsErr bool
+		name       string
+	}{
+		{name: "Divide ones", a: 1, b: 1, rest: []float64{}, want: 1},
+		{name: "Divide 2 by 1", a: 2, b: 1, rest: []float64{}, want: 2},
+		{name: "Divide 1 by 2", a: 1, b: 2, rest: []float64{}, want: 0.5},
+		{name: "Divide positive by zero", a: 1, b: 0, expectsErr: true},
+		{name: "Divide positive by zero", a: -1, b: 0, expectsErr: true},
+		{name: "Divide negative by zero", a: 0, b: 0, expectsErr: true},
+		{name: "Divide negative by zero", a: -1, b: -1, rest: []float64{}, want: 1},
+		{name: "Divide positive by negative", a: 2, b: -1, rest: []float64{}, want: -2},
+		{name: "Divide 2 numbers", a: 1, b: 2, rest: []float64{}, want: .5},
+		{name: "Divide 3 numbers", a: 1, b: 2, rest: []float64{.5}, want: 1},
```

Comment:
What about a case where one of the `rest` numbers is zero? As the implementer, you know that you have separate logic for that, and therefore it's vulnerable. This is actually a great way to generate test cases: look at every line of your function and imagine how it could be wrong, and what input would demonstrate that!